components/coreapi/component.go

```diff
@@ -202,16 +215,140 @@ func configure() error {
 		return httpserver.JSONResponse(c, http.StatusOK, resp)
 	})
 
+	routeGroup.GET(RouteCommitmentByID, func(c echo.Context) error {
+		commitmentID, err := httpserver.ParseCommitmentIDParam(c, restapipkg.ParameterCommitmentID)
+		if err != nil {
+			return err
+		}
+		index := commitmentID.Index()
+
+		resp, err := getCommitment(index)
+		if err != nil {
+			return err
+		}
+
+		return httpserver.JSONResponse(c, http.StatusOK, resp)
+	}, checkNodeSynced())
+
+	routeGroup.GET(RouteCommitmentByIDUTXOChanges, func(c echo.Context) error {
+		_, err := httpserver.ParseCommitmentIDParam(c, restapipkg.ParameterCommitmentID)
+		if err != nil {
+			return err
+		}
+
+		// get commitment utxos
+		return httpserver.JSONResponse(c, http.StatusOK, &slotUTXOResponse{})
+	}, checkNodeSynced())
+
+	routeGroup.GET(RouteSlotByIndex, func(c echo.Context) error {
+		indexUint64, err := httpserver.ParseUint64Param(c, restapipkg.ParameterSlotIndex)
+		if err != nil {
+			return err
+		}
+
+		resp, err := getCommitment(iotago.SlotIndex(indexUint64))
+		if err != nil {
+			return err
+		}
+
+		return httpserver.JSONResponse(c, http.StatusOK, resp)
+	}, checkNodeSynced())
+
+	routeGroup.GET(RouteSlotByIndexUTXOChanges, func(c echo.Context) error {
+		_, err := httpserver.ParseUint64Param(c, restapipkg.ParameterSlotIndex)
+		if err != nil {
+			return err
+		}
+
+		// get utxo changes
+		return httpserver.JSONResponse(c, http.StatusOK, &slotUTXOResponse{})
+	}, checkNodeSynced())
+
+	routeGroup.GET(RouteOutput, func(c echo.Context) error {
+		mimeType, err := httpserver.GetAcceptHeaderContentType(c, httpserver.MIMEApplicationVendorIOTASerializerV1, echo.MIMEApplicationJSON)
+		if err != nil && err != httpserver.ErrNotAcceptable {
+			return err
+		}
+
+		output, err := getOutput(c)
+		if err != nil {
+			return err
+		}
+
+		// default to echo.MIMEApplicationJSON
+		switch mimeType {
+		case httpserver.MIMEApplicationVendorIOTASerializerV1:
+			return c.Blob(http.StatusOK, httpserver.MIMEApplicationVendorIOTASerializerV1, output.Bytes())
+
+		default:
+			j, err := deps.Protocol.API().JSONEncode(output.Output())
+			if err != nil {
+				return err
+			}
+
+			return c.Blob(http.StatusOK, echo.MIMEApplicationJSON, j)
+		}
```

Comment:
you can refactor this whole part (reading mimetype, switching over it, encoding an object and returning the response) into a function that takes an `any` argument so that you simply pass an output to, or a block, etc.

Reply:
that way you don't need to copy-paste it so many times